File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        // simple wala tarika for complexity o(nlogn)
         // Agar length alag hai, toh pehle hi false return kar do
        if (s.length() != t.length()) {
            return false;
        }
        
        // C++ me direct string ko sort kar sakte hain
        // begin() se end() tak alphabetically arrange ho jayega
        sort(s.begin(), s.end());
        sort(t.begin(), t.end());
        
        // Agar sorting ke baad string s aur string t ek jaisi hain toh true, nahi toh false
        return s == t;
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // simple 
           // Agar dono strings ki length hi alag hai, toh anagram ho hi nahi sakte
        if (s.length() != t.length()) {
            return false;
        }
        
        // Strings ko character array me badla taaki sort kar sakein
        char[] sChars = s.toCharArray(); // camelCase
        char[] tChars = t.toCharArray(); // camelCase
        
        // Alphabetical order (A to Z) me sort kiya
        Arrays.sort(sChars);
        Arrays.sort(tChars);
        
        // Check kiya ki kya sorting ke baad dono arrays barabar hain
        return Arrays.equals(sChars, tChars);
    }
}